PCSX2/pcsx2
# SPU2: Add Cubeb backend, remove Portaudio and SDL2 backends

### Description of Changes
Another change from the Qt branch.

Everything should be functional except device selection, which still needs to be added. But do we care? Linux didn't have it before.

Cubeb also handles device hotplugging quite nicely :)

The last commit adds a pause method for the device. Currently, this isn't used, but in Qt we call it when the VM is paused, since we don't tear down everything there, we don't want to be burning CPU cycles/battery when we're paused.

Currently the Windows default is still XAudio. Did we want to change this?

### Rationale behind Changes

Consistency across platforms. Windows used portaudio/xaudio, Linux used portaudio or sdl, and Mac used sdl. This puts all platforms on a level playing field, although I left the xaudio backend for Windows in place because it's not really hurting anything.

### Suggested Testing Steps

 - Make sure audio works (I've done this on Windows/Linux).
 - Make sure surround expansion works (I can't test this myself).
